### DISCUSSION TITLE
How do I remove all unused coordinates?

### DISCUSSION THREAD

**Discussion by cmdupuis3** [pydata/xarray#7465] [Q&A] [answered]
I have an xarray dataset. It has about 50 different coordinates, and I'm only using three of them max. How do I remove all the coordinates that aren't used by any data arrays?

**Comment by TomNicholas** [comment, marked as answer]
Coordinates are not "used" by data variables, so I'm not entirely sure what you mean.

Variables depend on dimensions, but coordinates are a separate special set of variables. Coordinates may have the same name as dimensions though, and if they are one-dimensional they are referred to as "dimension coordinates". I appreciate that that is confusing.

If you just want to remove all the coordinates that aren't dimension coordinates, you could do 

```python
ds.drop_vars([var for var in ds.coords if var not in ds.dims])
```
